src/transformers/models/patchtsmixer/modeling_patchtsmixer.py

```diff
@@ -0,0 +1,2362 @@
+            than 1.
+
+        target_values (`torch.FloatTensor` of shape `(batch_size, target_len, input_size)` for forecasting,
+            `(batch_size, n_targets)` for regression, or `(batch_size,)` for classification, *optional*): Target values
+            of the time series, that serve as labels for the model. The `target_values` is what the Transformer needs
+            during training to learn to output, given the `context_values`. Note that, this is NOT required for a
+            pretraining task.
+
+            For a forecasting task, the shape is be `(batch_size, target_len, input_size)`. Even if we want to forecast
+            only specific channels by setting the indices in `forecast_channel_indices` parameter, pass the target data
+            with all channels, as channel Filtering for both prediction and target will be manually applied before the
+            loss computation.
+
+            For a classification task, it has a shape of `(batch_size,)`.
+
+            For a regression task, it has a shape of `(batch_size, n_targets)`.
+
+        output_hidden_states (`bool`, *optional*):
+            Whether or not to return the hidden states of all layers.
+
+        return_loss (`bool`,  *optional*):
+            Whether to return the loss in the `forward` call.
+"""
+
+
+class PatchTSMixerGatedAttention(nn.Module):
+    """PatchTSMixerGatedAttention
+
+    Args:
+        in_size (`int`): input size
+        out_size (`int`): output size
+    """
+
+    def __init__(self, in_size: int, out_size: int):
+        super().__init__()
+        self.attn_layer = nn.Linear(in_size, out_size)
+        self.attn_softmax = nn.Softmax(dim=-1)
+
+    def forward(self, inputs):
+        attn_weight = self.attn_softmax(self.attn_layer(inputs))
+        inputs = inputs * attn_weight
+        return inputs
+
+
+class PatchTSMixerTranspose(nn.Module):
+    """
+    Transpose the tensor to the dimension defined in **dims**
+    
+    Args:
+        dims (`list`, *optional*, defaults to `False`):
+               List of dimensions to be transposed contiguous. if True, the transposed tensor is contiguous.
+    """
+
+    def __init__(self, *dims, contiguous=False):
+        super().__init__()
+        self.dims, self.contiguous = dims, contiguous
+
+    def forward(self, inputs: torch.Tensor):
+        """
+        Parameters:
+            inputs (`torch.Tensor`): input to be transposed
+        Returns:
+            `torch.Tensor`: transposed tensor
+        """
+        if self.contiguous:
+            return inputs.transpose(*self.dims).contiguous()
+        else:
+            return inputs.transpose(*self.dims)
+
+
+class PatchTSMixerNormLayer(nn.Module):
+    def __init__(
+        self,
+        norm_mlp="LayerNorm",
+        mode="common_channel",
+        num_features=16,
+    ):
+        super().__init__()
+        self.norm_mlp = norm_mlp
```

Comment:
if only layernorm is used, I'd remove the norm_mlp argument and just use nn.LayerNorm here

Reply:
BatchNorm is also used, as can be seen below the marked lines.